docs/api-documentation.md

```diff
@@ -0,0 +1,370 @@
+   - It interacts with the `UserRepository` to handle user-related database operations.
+
+5. **UserRepository Database Interaction:**
+
+   - The `UserRepository` interacts with the database (Entity Framework) to:
+     - Create a new `IdentityApplicationUser` (an identity entity).
+     - Use `CreateAsync` method of `UserManager` Component to create a new identity user
+     - Add user details to the `ApplicationUser` entity.
+     - Associate user roles.
+     - Commit the changes to the database within a transaction.
+
+6. **JWT Token Generation:**
+
+   - Upon successful registration, the `AuthenticationService` generates a JWT token using user information.
+
+7. **Response to Client:**
+
+   - The `AuthenticationController` responds to the client with the generated JWT token.
+
+8. **ASP.NET Identity Components Involved:**
+
+   - `IdentityApplicationUser`: Represents the user entity in the Identity system.
+   - `SignInManager<IdentityApplicationUser>`: Manages sign-in functionality.
+   - `UserManager<IdentityApplicationUser>`: Manages user-related operations.
+   - `RoleManager<IdentityRole>`: Manages roles in the Identity system.
+   - `JwtSecurityToken`: Represents a JWT token.
+   - `SqlDatabaseContext`: Represents the database context for Entity Framework.
+
+The overall flow ensures a secure and structured user registration process using key components provided by ASP.NET, including Identity for user management and Entity Framework for database interactions. The separation of concerns among the controller, service, and repository ensures maintainability and scalability of the authentication process.
+
+**Database Table**
+
+Asp.Net User Table
+![Asp.Net User Table](./images/db_table_aspnetusers.png)
+
+User Table
+
+![User Table](./images/db_table_users.png)
+
+User Role
+
+![User Role](./images/db_table_userrole.png)
+
+**Endpoint:** `http://localhost:5000/api/auth/register`
+
+**Method: `POST`**
+
+**Body:**
+
+```json
+{
+  "firstName": "firstname",
+  "lastName": "lastname",
+  "email": "example@gmail.com",
+  "password": "Change@123"
+}
+```
+
+**Curl**
+
+```sh
+curl -X 'POST' \
+  'http://localhost:5000/api/auth/register' \
+  -H 'accept: */*' \
+  -H 'Content-Type: application/json' \
+  -d '{
+  "firstName": "firstname",
+  "lastName": "lastname",
+  "email": "example@gmail.com",
+  "password": "Change@123"
+}'
+```
+
+**Sample response**
+
+```json
+{
+  "token": "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJodHRwOi8vc2NoZW1hcy54bWxzb2FwLm9yZy93cy8yMDA1LzA1L2lkZW50aXR5L2NsYWltcy9lbWFpbGFkZHJlc3MiOiJkNzVhMmUzZS0yNmYxLTQzYTktYWZlOC00NWE0ZmZlOTljZDIiLCJodHRwOi8vc2NoZW1hcy54bWxzb2FwLm9yZy93cy8yMDA1LzA1L2lkZW50aXR5L2NsYWltcy9uYW1lIjoiaGFyaXNoQHNlLmNvbSIsImh0dHA6Ly9zY2hlbWFzLm1pY3Jvc29mdC5jb20vd3MvMjAwOC8wNi9pZGVudGl0eS9jbGFpbXMvcm9sZSI6IkRFRkFVTFQiLCJleHAiOjE3MDg1NDQ0NTYsImlzcyI6ImxvY2FsaG9zdCIsImF1ZCI6ImxvY2FsaG9zdCJ9.o5_1rncKAsq4aERSNgS37-3_726nyWrBDUNzXZFYL8Q"
+}
+```
+
+### Forget Password
```

Comment:
It's "Forgot password" or "Reset password"

Reply:
It refer to the forget password endpoint